data-prepper-plugins/s3-source/src/main/java/org/opensearch/dataprepper/plugins/source/s3/ScanObjectWorker.java

```diff
@@ -90,6 +93,7 @@ public ScanObjectWorker(final S3Client s3Client,
         this.pluginMetrics = pluginMetrics;
         acknowledgementSetCallbackCounter = pluginMetrics.counter(ACKNOWLEDGEMENT_SET_CALLBACK_METRIC_NAME);
         this.sourceCoordinator.initialize();
+        this.partitions = new ArrayList<>();
 
         this.partitionCreationSupplier = new S3ScanPartitionCreationSupplier(s3Client, bucketOwnerProvider, scanOptionsBuilderList, s3ScanSchedulingOptions);
```

Comment:
Nit: This class can be created only once and share between all the threads. Having it as it is know won't cause any issues though, just some extra instantiations.

Reply:
You mean the ScanObjectWorker class? We need a Runnable class for each thread. Not using this means we need to define another runnable class, right?

Reply:
Not the S3ScanPartitionCreationSupplier class. We could only instantiate it once and pass to all of the ScanObjectWorkers